File: infrastructure/db-core/src/main/java/com/nmh/commerce/utils/MoneyConverter.java
package com.nmh.commerce.utils;

import com.nmh.commerce.domain.Money;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Convert;
import jakarta.persistence.Converter;

import java.math.BigDecimal;

@Converter(autoApply = true)
public class MoneyConverter implements AttributeConverter<Money, BigDecimal> {
    @Override
    public BigDecimal convertToDatabaseColumn(Money money) {
        if (money == null) {
            return null;
        }
        return money.getValue();
    }

    @Override
    public Money convertToEntityAttribute(BigDecimal bigDecimal) {
        if (bigDecimal == null) {
            return null;
        }
        return Money.of(bigDecimal);
    }
}

File: infrastructure/db-core/src/main/kotlin/com/nmh/commerce/utils/MoneyConverter.kt
package com.nmh.commerce.utils

import com.nmh.commerce.domain.Money
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.math.BigDecimal

@Converter(autoApply = true)
class MoneyConverter : AttributeConverter<Money?, BigDecimal?> {
    override fun convertToDatabaseColumn(money: Money?): BigDecimal? {
        if (money == null) {
            return null
        }
        return money.value
    }

    override fun convertToEntityAttribute(bigDecimal: BigDecimal?): Money? {
        if (bigDecimal == null) {
            return null
        }
        return Money.of(bigDecimal)
    }
}
